### DISCUSSION TITLE
Can I control the table column width on listing pages?

### DISCUSSION THREAD

**Discussion by austin-hoover** [quarto-dev/quarto-cli#6155] [Q&A] [answered]
### Description

I would like to keep the date on one line.

<img width="841" alt="Screen Shot 2023-07-09 at 2 02 12 PM" src="https://github.com/quarto-dev/quarto-cli/assets/58012334/0df8ba7d-8b04-4e55-bbd4-5f5fcd1f56e2">

https://austin-hoover.github.io/notes/books/

**Comment by dragonstyle** [comment]
The HTML table is computing the column widths based upon the widest item (your longest title and author). Adding CSS to the listing page to force the date to be wider like:

```{=css}
<style>
.listing-date {
    width: 100px;
}
</style>
```

Should do it (you could also make a CSS file or add this to an existing CSS file).

**Reply by mcanouil** [reply]
Additionally, you could avoid word break for some columns:

```css
.listing-date {
  & a {
    word-break: normal;
  }
}
```

See <https://developer.mozilla.org/fr/docs/Web/CSS/word-break> for possible values.
Current value in Quarto is `word-break: keep-all;`.

**Reply by cscheid** [reply]
@dragonstyle @rich-iannone I wonder if we could detect that a column is entirely made of dates, and then add a class that has `word-break: normal` by default.

**Reply by rich-iannone** [reply, marked as answer]
@cscheid That's a good idea. Another option might be to use non-breaking hyphens (`#8209;`) in place of ASCII hyphens when `dateStyle` is `"YYYY-MM-DD"`. I tested this in a listing page and the date definitely does not break across lines.

**Reply by cscheid** [reply]
Huh - I like that one even better, because we probably know the date style and can use `#8209;` when appropriate..

**Reply by dragonstyle** [reply]
That is a fun idea- in the meantime we do support `date-format` as an option under `listing` as well - so you can provide a date format yourself that uses a different (non-wrapping) separator.

**Comment by SHogenboom** [comment]
Is there a more general solution that allows you to set the width of each of the columns?

My current listing:
![example-listing](https://github.com/quarto-dev/quarto-cli/assets/28568743/570d2215-d6d7-403b-89fc-a02453d4b029)

I would like to tweak the width of the first column so that the titles are easily readable. This would obviously reduce the width of the second column. Then for the third column I wouldn’t mind the styling, though a list (because they are categories) would perhaps be preferred.

Any help is appreciated!

**Reply by dragonstyle** [reply]
One idea would be to use CSS to target the column, something like:

```css
table td:nth-child(1)  {
  width: 100px;
}
```

or perhaps

```css
table td:nth-child(1)  {
  white-space: nowrap;
}

```
